app/src/main/java/com/example/sharingang/items/ItemsAdapter.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.sharingang.items
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.sharingang.R
+
+class ItemsAdapter : RecyclerView.Adapter<ItemViewHolder>() {
+    var data = listOf<Item>()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val view = layoutInflater.inflate(R.layout.item_view, parent, false) as TextView
+        return ItemViewHolder(view)
+    }
```

Comment:
How does does this method (`onCreateViewHolder`) work? I am not sure I understand its functionality

Reply:
So, for a list of different items, each of the items you display needs to create a view / widget on the screen. Because of this, you have `ItemViewHolder`, which encapsulates this widget. Basically, this section of code creates this widget dynamically, using the description in `item_view.xml`, which describes a single `TextView` widget.

You need to get a `LayoutInflater` from whatever surrounding context you have, in order to take the description of the layout for this widget in the xml file, and actually create a widget on the screen.

Reply:
Perfectly clear, thanks!